### PR TITLE
Use OTel recommended names for attributes

### PR DESCRIPTION
**Pull request by Cantido**
OpenTelemetry documents some conventional names to use for some pieces of GraphQL server data. This change renames one attribute and adds two more in line with OTel's recommendation, and also changes the name given to the span that is started.

## Added

- Attribute `graphql.operation.name` was added
- Attribute `graphql.operation.type` was added

## Changed
- BREAKING: `graphql.request.query` was renamed to `graphql.document`.
- Span name was changed to `graphql.operation.type` + `graphql.operation.name` if those fields are present, else "GraphQL Operation"

See: <https://opentelemetry.io/docs/reference/specification/trace/semantic_conventions/instrumentation/graphql/>

### CODE REVIEW

File: lib/instrumentation.ex [hunk 1]
```diff
@@ -56,18 +59,30 @@ defmodule OpentelemetryAbsinthe.Instrumentation do
 
   def handle_operation_start(_event_name, _measurements, metadata, config) do
     params = metadata |> Map.get(:options, []) |> Keyword.get(:params, %{})
+    document = params["query"]
+    operation_name = params["operationName"]
```
Comment (cpiemontese) [inline lib/instrumentation.ex:63]: I wonder if these should be `Map.get`s, direct access makes me nervous but maybe I'm just paranoid

File: lib/instrumentation.ex [hunk 2]
```diff
@@ -56,18 +59,30 @@ defmodule OpentelemetryAbsinthe.Instrumentation do
...
     attributes =
       []
       |> put_if(
         config.trace_request_variables,
         {"graphql.request.variables", Jason.encode!(params["variables"])}
       )
-      |> put_if(config.trace_request_query, {"graphql.request.query", params["query"]})
+      |> put_if(config.trace_request_query, {Atom.to_string(Conventions.graphql_document()), document})
+      |> put_if(config.trace_request_name, {Atom.to_string(Conventions.graphql_operation_name()), operation_name})
+      |> put_if(config.trace_request_type, {Atom.to_string(Conventions.graphql_operation_type()), operation_type})
```
Comment (cpiemontese) [inline lib/instrumentation.ex:74]: This looks a bit busy with all the conversions, maybe they could be moved to separate variables, wdyt?

File: lib/instrumentation.ex [hunk 3]
```diff
@@ -56,18 +59,30 @@ defmodule OpentelemetryAbsinthe.Instrumentation do
...
+
+    span_name =
+      cond do
+        not is_nil(operation_type) and not is_nil(operation_name) -> "#{operation_type} #{operation_name}"
+        not is_nil(operation_type) -> operation_type
+        true -> config.span_name
+      end
```
Comment (cpiemontese) [inline lib/instrumentation.ex:81]: Time for a `(get|build|...)_span_name` function?

**Review by cpiemontese — APPROVED** [review]

**Comment by cpiemontese** [comment]
Thank you very much for this PR :pray:

I added a few comments but mistakingly approved so ignore that :sweat_smile:

**Review by MaeIsBad — COMMENTED** [review]
Hi! Sorry for taking a while to review, the repo didn't have a CODEOWNERS(fixed https://github.com/primait/opentelemetry_absinthe/pull/84), so we missed the PR.

We agree that this is a great change and want this merged, but while reviewing I discovered that our master branch is actually quite broken(https://github.com/primait/opentelemetry_absinthe/pull/83). I'd like to get that merged first and then rebase your work on that.

Thanks for the contribution 💙

File: lib/instrumentation.ex (hunk 1 above)
Comment (MaeIsBad) [reply lib/instrumentation.ex:63]: Keyword.get returns nil if the map key is undefined so I think this is fine. 

too much :crab:?

File: lib/instrumentation.ex (hunk 2 above)
Comment (Cantido) [reply lib/instrumentation.ex:74]: Yeah, I just realized that I don't exactly need the `Atom.to_string/1` part, so I'll take that off and see if it looks any better.

File: lib/instrumentation.ex (hunk 3 above)
Comment (Cantido) [reply lib/instrumentation.ex:81]: Not a bad idea!

File: lib/instrumentation.ex (hunk 1 above)
Comment (cpiemontese) [reply lib/instrumentation.ex:63]: Also TIL `nil["a"] = nil` apparently, so fine I guess? :laughing: Too much JS programming left me paranoid

File: lib/instrumentation.ex [hunk 4]
```diff
@@ -111,6 +126,16 @@ defmodule OpentelemetryAbsinthe.Instrumentation do
     :ok
   end
 
+  defp get_operation_type(nil), do: nil
+  defp get_operation_type(document) do
```
Comment (MaeIsBad) [inline lib/instrumentation.ex:130]: I can thing of a lot of ways to break this implementation(whitespace, complex queries, first letter uppercase, other exotic LangaugeAdapters), if we move this to `handle_operation_stop` we can instead make this more robust by doing 
```elixir
data.blueprint |> Absinthe.Blueprint.current_operation() |> Kernel.||(%{}) |> Map.get(:type)
```

**Review by MaeIsBad — COMMENTED** [review]
You PR description contains a well written changelog, would you mind putting that in the unreleased section of CHANGELOG.md?

**Comment by MaeIsBad** [comment]
Rebased on master.

**Review by cpiemontese — APPROVED** [review]

**Comment by MaeIsBad** [comment]
It is, thank you 🙏

**Pull request merged**